From 68ca0888cfae3feeb0d7c545f80d4e81be7b808d Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Wed, 19 Sep 2012 14:45:37 +0800
Subject: [PATCH 597/609] arm: omap: am33xx: give the more compatibility with
 other omap boards

This patch tries to fix compiling errors for co-existing with other omap boards.

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm/mach-omap2/clock33xx.h   |    6 +++++-
 arch/arm/mach-omap2/clockdomain.h |    4 ++++
 arch/arm/mach-omap2/cminst44xx.h  |   11 +++++++++--
 arch/arm/mach-omap2/io.c          |    2 ++
 arch/arm/mach-omap2/powerdomain.h |    4 ++++
 arch/arm/mach-omap2/prminst44xx.h |   12 ++++++++++++
 arch/arm/mach-omap2/voltage.h     |    4 ++++
 arch/arm/plat-omap/sram.c         |    4 ++++
 8 files changed, 44 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-omap2/clock33xx.h b/arch/arm/mach-omap2/clock33xx.h
index 8409288..36d00ff 100644
--- a/arch/arm/mach-omap2/clock33xx.h
+++ b/arch/arm/mach-omap2/clock33xx.h
@@ -20,7 +20,11 @@
 #define AM33XX_MAX_DPLL_DIV   128
 
 
-int am33xx_clk_init(void);
+#ifdef CONFIG_SOC_AM33XX
+extern int am33xx_clk_init(void);
+#else
+static inline int am33xx_clk_init(void) {return 0;}
+#endif
 
 /* TRM ERRATA: Timer 3 & 6 default parent (TCLKIN) may not be always
      physically present, in such a case HWMOD enabling of
diff --git a/arch/arm/mach-omap2/clockdomain.h b/arch/arm/mach-omap2/clockdomain.h
index 72cb12b..a9fa541 100644
--- a/arch/arm/mach-omap2/clockdomain.h
+++ b/arch/arm/mach-omap2/clockdomain.h
@@ -195,7 +195,11 @@ int clkdm_hwmod_disable(struct clockdomain *clkdm, struct omap_hwmod *oh);
 extern void __init omap242x_clockdomains_init(void);
 extern void __init omap243x_clockdomains_init(void);
 extern void __init omap3xxx_clockdomains_init(void);
+#ifdef CONFIG_SOC_AM33XX
 extern void __init am33xx_clockdomains_init(void);
+#else
+static inline void __init am33xx_clockdomains_init(void) {}
+#endif
 extern void __init omap44xx_clockdomains_init(void);
 extern void _clkdm_add_autodeps(struct clockdomain *clkdm);
 extern void _clkdm_del_autodeps(struct clockdomain *clkdm);
diff --git a/arch/arm/mach-omap2/cminst44xx.h b/arch/arm/mach-omap2/cminst44xx.h
index 95320ee..6c1aaac 100644
--- a/arch/arm/mach-omap2/cminst44xx.h
+++ b/arch/arm/mach-omap2/cminst44xx.h
@@ -17,9 +17,9 @@ extern void omap4_cminst_clkdm_disable_hwsup(u8 part, s16 inst, u16 cdoffs);
 extern void omap4_cminst_clkdm_force_sleep(u8 part, s16 inst, u16 cdoffs);
 extern void omap4_cminst_clkdm_force_wakeup(u8 part, s16 inst, u16 cdoffs);
 
-extern int omap4_cminst_wait_module_ready(u8 part, u16 inst, s16 cdoffs, u16 clkctrl_offs);
 
 # if defined (CONFIG_ARCH_OMAP4) || defined (CONFIG_SOC_AM33XX)
+extern int omap4_cminst_wait_module_ready(u8 part, u16 inst, s16 cdoffs, u16 clkctrl_offs);
 extern int omap4_cminst_wait_module_idle(u8 part, u16 inst, s16 cdoffs,
 					 u16 clkctrl_offs);
 
@@ -28,7 +28,13 @@ extern void omap4_cminst_module_enable(u8 mode, u8 part, u16 inst, s16 cdoffs,
 extern void omap4_cminst_module_disable(u8 part, u16 inst, s16 cdoffs,
 					u16 clkctrl_offs);
 
+extern void __init omap44xx_cminst_init(void);
 # else
+static inline int omap4_cminst_wait_module_ready(u8 part, u16 inst, s16 cdoffs,
+						 u16 clkctrl_offs) 
+{
+	return 0;
+}
 
 static inline int omap4_cminst_wait_module_idle(u8 part, u16 inst, s16 cdoffs,
 					u16 clkctrl_offs)
@@ -46,6 +52,8 @@ static inline void omap4_cminst_module_disable(u8 part, u16 inst, s16 cdoffs,
 {
 }
 
+static inline void __init omap44xx_cminst_init(void) {}
+
 # endif
 
 /*
@@ -63,5 +71,4 @@ extern u32 omap4_cminst_clear_inst_reg_bits(u32 bits, u8 part, s16 inst,
 extern u32 omap4_cminst_read_inst_reg_bits(u8 part, u16 inst, s16 idx,
 					   u32 mask);
 
-extern void __init omap44xx_cminst_init(void);
 #endif
diff --git a/arch/arm/mach-omap2/io.c b/arch/arm/mach-omap2/io.c
index a1f4549..8abfb76 100644
--- a/arch/arm/mach-omap2/io.c
+++ b/arch/arm/mach-omap2/io.c
@@ -469,6 +469,7 @@ void __init ti81xx_init_early(void)
 	omap3xxx_clk_init();
 }
 
+#ifdef CONFIG_SOC_AM33XX
 void __init am33xx_init_early(void)
 {
 	omap2_set_globals_am33xx();
@@ -485,6 +486,7 @@ void __init am33xx_init_early(void)
 	omap3xxx_clk_init();
 }
 #endif
+#endif
 
 #ifdef CONFIG_ARCH_OMAP4
 void __init omap4430_init_early(void)
diff --git a/arch/arm/mach-omap2/powerdomain.h b/arch/arm/mach-omap2/powerdomain.h
index 437e51d..6c6f116 100644
--- a/arch/arm/mach-omap2/powerdomain.h
+++ b/arch/arm/mach-omap2/powerdomain.h
@@ -227,7 +227,11 @@ bool pwrdm_can_ever_lose_context(struct powerdomain *pwrdm);
 extern void omap242x_powerdomains_init(void);
 extern void omap243x_powerdomains_init(void);
 extern void omap3xxx_powerdomains_init(void);
+#ifdef CONFIG_SOC_AM33XX
 extern void am33xx_powerdomains_init(void);
+#else
+static inline void am33xx_powerdomains_init(void) {}
+#endif
 extern void omap44xx_powerdomains_init(void);
 
 extern struct pwrdm_ops omap2_pwrdm_operations;
diff --git a/arch/arm/mach-omap2/prminst44xx.h b/arch/arm/mach-omap2/prminst44xx.h
index 53a4b45..0bf2618 100644
--- a/arch/arm/mach-omap2/prminst44xx.h
+++ b/arch/arm/mach-omap2/prminst44xx.h
@@ -23,6 +23,7 @@ extern u32 omap4_prminst_rmw_inst_reg_bits(u32 mask, u32 bits, u8 part,
 
 extern void omap4_prminst_global_warm_sw_reset(void);
 
+#if defined (CONFIG_ARCH_OMAP4) || defined(CONFIG_SOC_AM33XX)
 extern int omap4_prminst_is_hardreset_asserted(u8 shift, u8 part, s16 inst,
 					       u16 rstctrl_offs);
 extern int omap4_prminst_assert_hardreset(u8 shift, u8 part, s16 inst,
@@ -30,4 +31,15 @@ extern int omap4_prminst_assert_hardreset(u8 shift, u8 part, s16 inst,
 extern int omap4_prminst_deassert_hardreset(u8 shift, u8 st_shift, u8 part,
 				s16 inst, u16 rstctrl_offs, u16 rstst_offs);
 extern void __init omap44xx_prminst_init(void);
+#else
+static inline int omap4_prminst_is_hardreset_asserted(u8 shift, u8 part, s16 inst,
+					       u16 rstctrl_offs) {return 0;}
+static inline int omap4_prminst_assert_hardreset(u8 shift, u8 part, s16 inst,
+					  u16 rstctrl_offs) {return 0;}
+static inline int omap4_prminst_deassert_hardreset(u8 shift, u8 st_shift, u8 part,
+				s16 inst, u16 rstctrl_offs, u16 rstst_offs)
+				{return 0;}
+static inline void __init omap44xx_prminst_init(void) {}
+#endif
+
 #endif
diff --git a/arch/arm/mach-omap2/voltage.h b/arch/arm/mach-omap2/voltage.h
index a7c43c1..c4d3d72 100644
--- a/arch/arm/mach-omap2/voltage.h
+++ b/arch/arm/mach-omap2/voltage.h
@@ -156,7 +156,11 @@ int omap_voltage_late_init(void);
 
 extern void omap2xxx_voltagedomains_init(void);
 extern void omap3xxx_voltagedomains_init(void);
+#ifdef CONFIG_SOC_AM33XX
 extern void am33xx_voltagedomains_init(void);
+#else
+static inline void am33xx_voltagedomains_init(void) {}
+#endif
 extern void omap44xx_voltagedomains_init(void);
 
 struct voltagedomain *voltdm_lookup(const char *name);
diff --git a/arch/arm/plat-omap/sram.c b/arch/arm/plat-omap/sram.c
index 4976758..e13adbb 100644
--- a/arch/arm/plat-omap/sram.c
+++ b/arch/arm/plat-omap/sram.c
@@ -355,11 +355,15 @@ static inline int omap34xx_sram_init(void)
 }
 #endif /* CONFIG_ARCH_OMAP3 */
 
+#ifdef CONFIG_SOC_AM33XX
 static inline int am33xx_sram_init(void)
 {
 	am33xx_push_sram_idle();
 	return 0;
 }
+#else
+static inline int am33xx_sram_init(void) {return 0;}
+#endif
 
 int __init omap_sram_init(void)
 {
-- 
1.7.5.4

